source/base/polynomials_bdm.cc

```diff
@@ -28,7 +28,7 @@ DEAL_II_NAMESPACE_OPEN
 
 template <int dim>
 PolynomialsBDM<dim>::PolynomialsBDM(const unsigned int k)
-  : TensorPolynomialsBase<dim>(k, n_polynomials(k))
+  : TensorPolynomialsBase<dim>(k + 1, n_polynomials(k))
```

Comment:
So just to be clear, this was the bug you're fixing, right?

Reply:
Yes this is the bug I'm fixing. This is because in the base class, we have
```
  /**
   * The highest polynomial degree of this functions represented by this object.
   */
  const unsigned int polynomial_degree;
```